### PR TITLE
[Transform] fix issue in TransformIndexerStateTests.testStopAtCheckpoint

### LINKED ISSUE #62996: [CI] TransformIndexerStateTests.testStopAtCheckpoint
**Build scan**:

https://gradle-enterprise.elastic.co/s/yatg7ybt2srg6

**Repro line**:

```
./gradlew ':x-pack:plugin:transform:test' --tests "org.elasticsearch.xpack.transform.transforms.TransformIndexerStateTests.testStopAtCheckpoint" \
  -Dtests.seed=A036379B9D4C4E02 \
  -Dtests.security.manager=true \
  -Dtests.locale=vi \
  -Dtests.timezone=Etc/GMT-12 \
  -Druntime.java=11
```

**Reproduces locally?**:

Yes

**Applicable branches**:

`master`

**Failure history**:

**Failure excerpt**:

```
org.elasticsearch.xpack.transform.transforms.TransformIndexerStateTests > testStopAtCheckpoint FAILED
    java.lang.AssertionError:
    Expected: a value equal to or greater than <1>
         but: <0> was less than <1>
        at __randomizedtesting.SeedInfo.seed([A036379B9D4C4E02:A6AC16D3A43F3513]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.transform.transforms.TransformIndexerStateTests.testStopAtCheckpoint(TransformIndexerStateTests.java:439)
```

### PR DESCRIPTION
**Pull request by hendrikmuhs**
fix a test issue by improving counting the number of times the deferred listener is called

fixes #62996

Note: The test failed for a certain seed that called `_stop_at_checkpoint=false` all 6 times. I run this test with `-Dtests.iters` and could not spot another corner case

### CODE REVIEW

File: x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java [hunk 1]
```diff
@@ -412,20 +412,29 @@ public void testStopAtCheckpoint() throws Exception {
...
 
             List<CountDownLatch> responseLatches = new ArrayList<>();
+            int timesStopAtCheckpointChanged = 0;
+            // default stopAtCheckpoint is false
+            boolean previousStopAtCheckpoint = false;
+
             for (int i = 0; i < 3; ++i) {
                 CountDownLatch latch = new CountDownLatch(1);
                 boolean stopAtCheckpoint = randomBoolean();
+                timesStopAtCheckpointChanged = timesStopAtCheckpointChanged + (stopAtCheckpoint == previousStopAtCheckpoint ? 0 : 1);
```
Comment (benwtrent) [inline x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java:422]:
```suggestion
                timesStopAtCheckpointChanged += (stopAtCheckpoint == previousStopAtCheckpoint ? 0 : 1);
```

File: x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java [hunk 2]
```diff
@@ -412,20 +412,29 @@ public void testStopAtCheckpoint() throws Exception {
...
             searchLatch.countDown();
 
-            // this time call it 3 times
-            assertResponse(listener -> setStopAtCheckpoint(indexer, randomBoolean(), listener));
-            assertResponse(listener -> setStopAtCheckpoint(indexer, randomBoolean(), listener));
-            assertResponse(listener -> setStopAtCheckpoint(indexer, randomBoolean(), listener));
+            // call it 3 times again
+            for (int i = 0; i < 3; ++i) {
+                boolean stopAtCheckpoint = randomBoolean();
+                timesStopAtCheckpointChanged = timesStopAtCheckpointChanged + (stopAtCheckpoint == previousStopAtCheckpoint ? 0 : 1);
```
Comment (benwtrent) [inline x-pack/plugin/transform/src/test/java/org/elasticsearch/xpack/transform/transforms/TransformIndexerStateTests.java:434]:
```suggestion
                timesStopAtCheckpointChanged += (stopAtCheckpoint == previousStopAtCheckpoint ? 0 : 1);
```

**Review by benwtrent — APPROVED** [review]
minor syntax opinions.

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/packaging-sample-windows

**Comment by hendrikmuhs** [comment]
run elasticsearch-ci/packaging-sample-windows

**Pull request merged**